src/inttau2.f90

```diff
@@ -52,13 +52,19 @@ subroutine tauint2(grid, packet, sdfs_array)
             packet%cnts = packet%cnts + size(ds)
             d_sdf = minval(ds)
 
+            ! if(d_sdf < eps)then
+            !     packet%tflag=.true.
+            !     exit
+            ! end if
+
             do while(d_sdf > eps)
                 t_sdf = d_sdf * sdfs_array(packet%layer)%p%kappa
                 if(taurun + t_sdf <= tau)then
                     !move full distance to sdf surface
                     taurun = taurun + t_sdf
                     oldpos = pos
-                    call update_jmean(grid, oldpos, dir, d_sdf, packet, sdfs_array(packet%layer)%p%mua)
```

Comment:
Its probably cleaner to have phasor be updated in update_jmean. Though maybe change of name from update_jmean to update_grids? If you have a better name then please use it